src/calendar/model/CalendarModel.ts

```diff
@@ -801,6 +804,17 @@ export class CalendarModel {
 	getFileIdToSkippedCalendarEventUpdates(): Map<Id, CalendarEventUpdate> {
 		return this.fileIdToSkippedCalendarEventUpdates
 	}
+
+	canFullyEditEvent(event: CalendarEvent): boolean {
+		const userController = this.logins.getUserController()
+		const userMailGroup = userController.getUserMailGroupMembership().group
+		const mailboxDetailsArray = this.mailModel.mailboxDetails()
+		const mailboxDetails = assertNotNull(mailboxDetailsArray.find((md) => md.mailGroup._id === userMailGroup))
+		const ownMailAddresses = getEnabledMailAddressesWithUser(mailboxDetails, userController.userGroupInfo)
+		const calendarInfos = this.getCalendarInfosStream()()
+		const eventType = getEventType(event, calendarInfos, ownMailAddresses, userController.user)
+		return eventType === EventType.OWN || eventType === EventType.SHARED_RW
```

Comment:
We should probably document the method a bit better and mention that it mirrors the logic of  `CalendarEventModel.prototype.isFullyWritable()`